packages/in_app_purchase/in_app_purchase_ios/ios/Classes/FIAPReceiptManager.m

```diff
@@ -9,18 +9,22 @@ @implementation FIAPReceiptManager
 
 - (NSString *)retrieveReceiptWithError:(FlutterError **)error {
   NSURL *receiptURL = [[NSBundle mainBundle] appStoreReceiptURL];
-  NSData *receipt = [self getReceiptData:receiptURL];
+  NSError *err;
+  NSData *receipt = [self getReceiptData:receiptURL error:&err];
+  if (err) {
+    *error = [FlutterError errorWithCode:[[NSString alloc] initWithFormat:@"%li", (long)err.code]
+                                 message:err.domain
+                                 details:err.userInfo];
+    return nil;
+  }
   if (!receipt) {
-    *error = [FlutterError errorWithCode:@"storekit_no_receipt"
-                                 message:@"Cannot find receipt for the current main bundle."
-                                 details:nil];
```

Comment:
Why was this removed? There's now a codepath that returns nil but doesn't set the error.

Reply:
https://developer.apple.com/documentation/foundation/nsdata/1547238-datawithcontentsofurl

> This method may fail and return nil for various reasons, like bad URLs and file-not-found errors. In this case, the reason for error is indicated by errorPtr.

If this is nil it should always have given an error. This `if(!receipt)` can be removed completely, but that might give nullpointer if somehow the StoreKit does give nil without an error. I added an error, but I'm also fine with removing it completely.

Reply:
Ah, then I would just simplify this down to:
`if (!receipt || receiptError) {...}`
with the first error construction code. If receiptError is nil it'll still work, just not give any extra info (which is fine for a case that is documented not to happen).